### ISSUE TITLE
Mixed dtype data causes error in `pd.cut`

### ISSUE THREAD

**Issue by capnahab** [lux-org/lux#394]
I am not experienced in data analysis, - I am a surgeon looking at operation data for pituitary surgery. I am using a macbook pro with apple M 'silicon' processor.

Lux will not load after importing a csv in jupyter notebook. The csv is 62 fields and 6000 rows but fairly simple with some datetimes in about 6 fields. Defaults back to Pandas. 

I have the jupyter nbextension luxwidget in my extensions tab and no problems reported on 
jupyter nbextension install --py luxwidget
jupyter nbextension enable --py luxwidget

to reproduce. 

`import` pandas as pd
import lux

df = pd.read_csv("/Users/Desktop/ND/June_20/data/with.csv")
df[['ADMIDATE', 'DISDATE', 'EPIEND', 'EPISTART', 'OPDATE_01','DOD']] = df[['ADMIDATE', 'DISDATE', 'EPIEND', 'EPISTART', 'OPDATE_01','DOD']].apply(pd.to_datetime, dayfirst=True, errors='coerce')
df['fyear'] = pd.to_datetime(df['fyear'], format='%Y')

pit = ["B041", "B012"]
df_p = df[df["OPERTN_01"].isin(pit)].copy()
df_p

/opt/homebrew/lib/python3.9/site-packages/IPython/core/interactiveshell.py:3169: DtypeWarning:Columns (14,20,37,38) have mixed types.Specify dtype option on import or set low_memory=False.
/opt/homebrew/lib/python3.9/site-packages/IPython/core/formatters.py:918: UserWarning:
Unexpected error in rendering Lux widget and recommendations. Falling back to Pandas display.
Please report the following issue on Github: https://github.com/lux-org/lux/issues 

/opt/homebrew/lib/python3.9/site-packages/lux/core/frame.py:628: UserWarning:Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.9/site-packages/lux/core/frame.py", line 590, in _ipython_display_
    self.maintain_recs()
  File "/opt/homebrew/lib/python3.9/site-packages/lux/core/frame.py", line 432, in maintain_recs
    custom_action_collection = custom_actions(rec_df)
  File "/opt/homebrew/lib/python3.9/site-packages/lux/action/custom.py", line 76, in custom_actions
    recommendation = lux.config.actions[action_name].action(ldf)
  File "/opt/homebrew/lib/python3.9/site-packages/lux/action/correlation.py", line 50, in correlation
    vlist = VisList(intent, ldf)
  File "/opt/homebrew/lib/python3.9/site-packages/lux/vis/VisList.py", line 43, in __init__
    self.refresh_source(self._source)
  File "/opt/homebrew/lib/python3.9/site-packages/lux/vis/VisList.py", line 336, in refresh_source
    lux.config.executor.execute(self._collection, ldf, approx=approx)
  File "/opt/homebrew/lib/python3.9/site-packages/lux/executor/PandasExecutor.py", line 143, in execute
    PandasExecutor.execute_2D_binning(vis)
  File "/opt/homebrew/lib/python3.9/site-packages/lux/executor/PandasExecutor.py", line 422, in execute_2D_binning
    vis._vis_data["yBin"] = pd.cut(vis._vis_data[y_attr], bins=lux.config.heatmap_bin_size)
  File "/opt/homebrew/lib/python3.9/site-packages/pandas/core/reshape/tile.py", line 238, in cut
    rng = (nanops.nanmin(x), nanops.nanmax(x))
  File "/opt/homebrew/lib/python3.9/site-packages/pandas/core/nanops.py", line 135, in f
    result = alt(values, axis=axis, skipna=skipna, **kwds)
  File "/opt/homebrew/lib/python3.9/site-packages/pandas/core/nanops.py", line 394, in new_func
    result = func(values, axis=axis, skipna=skipna, mask=mask, **kwargs)
  File "/opt/homebrew/lib/python3.9/site-packages/pandas/core/nanops.py", line 977, in reduction
    result = getattr(values, meth)(axis)
  File "/opt/homebrew/lib/python3.9/site-packages/numpy/core/_methods.py", line 43, in _amin
    return umr_minimum(a, axis, None, out, keepdims, initial, where)
TypeError: '<=' not supported between instances of 'int' and `'str'`

**Comment by dorisjlee** [comment]
Hi @capnahab, Thank you for reporting this issue! This is likely happening because one of the columns that you are working with has multiple data types mixed in a single column (similar to an issue that we fixed #168). I can look into fixing this issue, but I don't have access to your data, so I am not able to reproduce the error that your seeing. 

Could you share a sample of the dataset so that we can reproduce the error for testing? (You can also email it to me at dorisjunglinlee@gmail.com if you prefer not to share the data in the public forum).
Happy to work with you to see how we can get this to work on your dataset!

**Comment by capnahab** [comment]
You are very kind. Unfortunately the data has patient identifiers in so I cannot share it. However I have used a method to look at the columns with mixed data types.
I can try and give you some information but presumably I need to clean up my columns?.
Looking at my dataframe with df.info() I get 


`<class 'pandas.core.frame.DataFrame'>
Int64Index: 6550 entries, 85 to 557832
Data columns (total 61 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   Unnamed: 0        6550 non-null   int64         
 1   fyear             6550 non-null   int64         
 2   ADMIAGE           6550 non-null   int64         
 3   ADMIDATE          6550 non-null   datetime64[ns]
 4   ADMIMETH          6550 non-null   object        
 5   NNAP_CONSULT      6550 non-null   int64         
 6   DIAG_01           6550 non-null   object        
 7   DIAG_02           6106 non-null   object        
 8   DIAG_03           5300 non-null   object        
 9   DIAG_04           4244 non-null   object        
 10  DIAG_05           3332 non-null   object        
 11  DISDATE           6504 non-null   datetime64[ns]
 12  DISDEST           6550 non-null   int64         
 13  DISMETH           6550 non-null   int64         
 14  DOMPROC           173 non-null    object        
 15  ENCRYPTED_HESID   6550 non-null   object        
 16  EPIEND            6550 non-null   datetime64[ns]
 17  EPIKEY            6550 non-null   int64         
 18  EPIORDER          6550 non-null   int64         
 19  EPISTART          6550 non-null   datetime64[ns]
 20  MAINSPEF          6550 non-null   object        
 21  OPDATE_01         6550 non-null   datetime64[ns]
 22  OPDATE_02         6317 non-null   datetime64[ns]
 23  OPDATE_03         4516 non-null   datetime64[ns]
 24  OPDATE_04         3655 non-null   datetime64[ns]
 25  OPDATE_05         3050 non-null   datetime64[ns]
 26  OPERTN_01         6550 non-null   object        
 27  OPERTN_02         6317 non-null   object        
 28  OPERTN_03         4516 non-null   object        
 29  OPERTN_04         3655 non-null   object        
 30  OPERTN_05         3050 non-null   object        
 31  POSTDIST          6505 non-null   object        
 32  PROCODE5          6550 non-null   object        
 33  SEX               6550 non-null   int64         
 34  SITETRET          6523 non-null   object        
 35  STARTAGE          6505 non-null   float64       
 36  SUSSPELLID        6550 non-null   float64       
 37  TRETSPEF          6550 non-null   object        
 38  DOD               420 non-null    datetime64[ns]
 39  opertn_01Type     6550 non-null   object        
 40  opertn_02Type     254 non-null    object        
 41  opertn_03Type     2365 non-null   object        
 42  opertn_04Type     1267 non-null   object        
 43  opertn_05Type     759 non-null    object        
 44  OPCS 4.9          6550 non-null   object        
 45  opcs_4.9str       6550 non-null   object        
 46  Description       6550 non-null   object        
 47  ICD10_str         5564 non-null   object        
 48  Code Description  5564 non-null   object        
 49  los               6504 non-null   float64       
 50  code_3            6550 non-null   object        
 51  code_5            6550 non-null   object        
 52  unit_name         6550 non-null   object        
 53  short_name        6550 non-null   object        
 54  NNAPID            6550 non-null   float64       
 55  GMC               6550 non-null   float64       
 56  GMCRefNo          6550 non-null   float64       
 57  Surname           6550 non-null   object        
 58  GivenName         6550 non-null   object        
 59  Gender            6550 non-null   object        
 60  Qualification     6550 non-null   object        
dtypes: datetime64[ns](10), float64(6), int64(9), object(36)
memory usage: 3.1+ MB` 


 I have used  the method described [here](https://stackoverflow.com/a/29376221/11795964).

This tells me that the floowing columns have data types that dont match the first row.

`DIAG_02
DIAG_03
DIAG_04
DIAG_05
DISDATE
DOMPROC
MAINSPEF
OPDATE_02
OPDATE_03
OPDATE_04
OPDATE_05
OPERTN_02
OPERTN_03
OPERTN_04
OPERTN_05
POSTDIST
SITETRET
TRETSPEF
DOD
opertn_02Type
opertn_03Type
opertn_04Type
opertn_05Type
ICD10_str
Code Description`

Is this enough information for you to help?.

**Comment by cgarciae** [comment]
@dorisjlee I am able to reproduce the error like this:

```python
a = [1, "2", 3, "4", 5, "6", 7, "8", 9, "10"] * 1_000
b = np.random.uniform(0, 1, size=len(a))

df = pd.DataFrame({"a": a, "b": b})

df._ipython_display_()
```

**Comment by dorisjlee** [comment]
Resolved this issue in #448. You can access these updated changes by upgrading to [the latest version of Lux](https://pypi.org/project/lux-api/):
```python
pip install --upgrade lux-api
jupyter nbextension install --py luxwidget
jupyter nbextension enable --py luxwidget
```
I hope that this resolves the problem that you were seeing!
